### PR TITLE
new_audit: csp-inline

### PR DESCRIPTION
**Pull request by adamraine**
Closes https://github.com/GoogleChrome/lighthouse/issues/13004

Needs https://developer.chrome.com/ doc before we can merge.

### CODE REVIEW

File: core/audits/csp-inline.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      requiredArtifacts: ['MainDocumentContent', 'MetaElements'],
+      supportedModes: ['navigation'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
```
Comment (connorjclark) [inline core/audits/csp-inline.js:44]: this doesn't need to be async

**Review by connorjclark — APPROVED** [review]

File: core/audits/csp-inline.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that advises users to avoid putting a CSP in an inline html meta tag. This descriptive title is shown to users when a CSP is not found in the page's html document. */
+  title: 'Does not define any CSPs in inline `<meta>` tags',
+  /** Title of a Lighthouse audit that advises users to avoid putting a CSP in an inline html meta tag. This descriptive title is shown to users when a CSP is found in the page's html document. */
+  failureTitle: 'Defines a CSP in an inline `<meta>` tag',
+  /** Description of a Lighthouse audit that advises users to avoid putting a CSP in an inline html meta tag. This is displayed after a user expands the section to see more. No character length limits. The last sentence starting with 'Learn' becomes link text to additional documentation. */
+  description: 'A CSP defined in an inline `<meta>` tag will delay the preload scanner from ' +
+    'loading resources early. Consider defining all CSPs in http headers if you can. ' +
+    '[Learn more about defining a CSP in an inline meta tag]()',
```
Comment (connorjclark) [inline core/audits/csp-inline.js:18]: todo: url

**Comment by brendankenny** [comment]
We should consider using `inline-csp` as more of a noun-y audit id, which would better fit the [naming guidelines](https://github.com/GoogleChrome/lighthouse/blob/main/docs/new-audits.md#naming)). I was against this in the sync just now, but now I kind of like it better :)

**Comment by adamraine** [comment]
Closing see https://github.com/GoogleChrome/lighthouse/issues/13004#issuecomment-1707328715

**Pull request closed without merging**